components/profile/pages/wallet/wallet-balance/controller.tsx

```diff
@@ -0,0 +1,194 @@
+    typeof token === "string" ? token : token?.address;
+
+  async function processToken(token: string | Token) {
+    const [tokenData, balance] = await Promise.all([
+      typeof token === "string"
+        ? state.Service?.active?.getERC20TokenData(token)
+        : token,
+      state.Service?.active?.getTokenBalance(getAddress(token),
+                                             state.currentUser.walletAddress),
+    ]);
+
+    const tokenInformation = await getCoinInfoByContract(tokenData.symbol);
+
+    return {
+      balance,
+      ...tokenData,
+      icon: <TokenIcon src={tokenInformation?.icon as string} />,
+    };
+  }
+
+  function updateSearch() {
+    setSearch(searchState);
+  }
+
+  function handleSearch(event) {
+    if (event.key !== "Enter") return;
+
+    updateSearch();
+  }
+
+  function handleClearSearch(): void {
+    setSearchState("");
+    setSearch("");
+  }
+
+  function handleSearchChange(e) {
+    setSearchState(e.target.value);
+    debouncedSearchUpdater(e.target.value);
+  }
+
+  function handleSearchFilter(name = "", symbol = "", networks) {
+    const hasNetworkName = query?.networkName
+    const isNetwork = !!networks.find(({ name }) =>
+      hasNetworkName?.toString().toLowerCase() === name?.toLowerCase());
+
+    return (
+      (name.toLowerCase().indexOf(search.toLowerCase()) >= 0 ||
+        symbol.toLowerCase().indexOf(search.toLowerCase()) >= 0) &&
+      (hasNetworkName ? isNetwork : true)
+    );
+  }
+
+  function loadBalances() {
+    if (!state.currentUser?.walletAddress) return;
+
+    searchCurators({
+      address: state.currentUser?.walletAddress,
+      chainShortName:
+        query?.chain?.toString() || state?.connectedChain?.shortName,
+    }).then(({ rows }) => {
+      Promise.all(rows?.map(async (curator) => {
+        const tokenInformation = await getCoinInfoByContract(curator?.network?.networkToken?.symbol);
+
+        return {
+            symbol: t("$oracles", {
+              token: curator?.network?.networkToken?.symbol,
+            }),
+            name: `${t("misc.locked")} ${curator?.network?.networkToken?.name}`,
+            address: curator?.network?.networkToken?.address,
+            icon: <TokenIcon src={tokenInformation?.icon as string} />,
+            oraclesLocked: BigNumber(curator.tokensLocked),
+            networkName: curator?.network?.name,
+        };
+      })).then(setTokensOracles);
+    });
+
+    if (state.Service?.starting) return;
+
+    getTokens(state?.connectedChain?.id).then((tokens) => {
+      Promise.all(tokens?.map(async (token) => {
+        const tokenData = await processToken(token?.address);
+        return { networks: token?.networks, ...tokenData };
+      })).then(setTokens);
+    });
+  }
+
+  useEffect(loadBalances, [
+    state.currentUser?.walletAddress,
+    state.connectedChain,
+    state.Service?.starting,
+  ]);
+
+  useEffect(() => {
+    if (!tokens?.length) return;
+
+    Promise.all(tokens.map(async (token) => ({
```

Comment:
align with @vhcsilva to come to a solution about merging this functionality into one x-hook because this is repeated on both PRs